homeassistant/components/ecobee/climate.py

```diff
@@ -726,6 +740,77 @@ def resume_program(self, resume_all):
         )
         self.update_without_throttle = True
 
+    def set_active_sensors(
+        self, sensors: list[str], climate_name: str | None = None
+    ) -> None:
+        """Set the currently active sensors on a thermostat."""
+        if climate_name is None:
+            climate_name = self.preset_mode
+
+        # Get device name from device id.
+        device_registry = dr.async_get(self.hass)
+        sensor_names: list[str] = []
+        for sensor in sensors:
+            sensor_name = device_registry.async_get(sensor).name
+            sensor_names.append(sensor_name)
+
+        # Check if climate is an available preset option.
+        if climate_name not in self._preset_modes.values():
+            _LOGGER.error(
+                f"Invalid climate name, available options are: "
+                f"{', '.join(self.preset_modes)}"
+            )
+            raise HomeAssistantError(
+                f"Invalid climate name, available options are: "
+                f"{', '.join(self.preset_modes)}"
+            )
+
+        # Ensure sensors provided are available for thermostat.
+        if not set(sensor_names).issubset(set(self._sensors)):
+            _LOGGER.error(
+                f"Invalid sensor for thermosat, available options are: "
+                f"{', '.join(self._sensors)}"
+            )
+            raise HomeAssistantError(
+                f"Invalid sensor for thermosat, available options are: "
+                f"{', '.join(self._sensors)}"
+            )
+
+        # Check if sensors are currently active on thermostat.
+        current_active_sensors = self.active_sensors(climate_name)
+        if set(sensor_names) == set(current_active_sensors):
+            _LOGGER.error(
```

Comment:
Same here

Reply:
This check uses information from the device registry lookup. I do not know how I would move it before it? Unless you are just talking about the msg format. In which case, I changed it.